### ISSUE TITLE
Compiling fails - error[E0599]

### ISSUE THREAD

**Issue by Openuser87** [cooklang/cooklang-rs#15]
Following error occurs upon using `cargo install cooklang-chef`:

```
error[E0599]: no variant or associated item named `Default` found for enum `Color` in the current scope
   --> /home/<user>/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cooklang-0.13.2/src/error.rs:496:20
    |
496 |             Color::Default
    |                    ^^^^^^^ variant or associated item not found in `Color`
```

Versions: cargo 1.77.2, rustc 1.77.2.

**Comment by Zheoni** [comment]
Thanks for reporting. This is an issue with the dependency `ariadne` that I already reported (https://github.com/zesterer/ariadne/issues/116). Until it's solved or a new version is released you can install `chef` with `cargo install cooklang-chef --locked`.

For library users, patch the `ariadne` version to `"=0.4.0"` until this is solved by, for example, adding it as a dependency even if you don't use it.

**Comment by johnhkchen** [comment]
Hello, it's been a few months but the `ariadne` crate is still inactive.

It seems like this other user switched to a different crate, https://github.com/01mf02/jaq/pull/185

https://crates.io/crates/codesnake

Codesnake is pretty explicitly meant as a rewrite of Ariadne.

Perhaps I can work on the switch to codesnake?

**Comment by Zheoni** [comment]
Hi, I have also been following the issue and considered `codesnake`. I started porting the code and it's pretty much done. But I realised that some features were missing or had limitations.

- As the `codesnake` description says `Pretty printer for non-overlapping code spans` *non-overlapping* and we have a couple of overlapping code spans. This should be easy to solve but requires manually checking every possible error span created by the compiler.

- It does not support "empty labels" which we use quite often. This is highlighting a piece of code without adding an arrow with a message. (Like the one in the image, the arrow should not be there)

![image](https://github.com/user-attachments/assets/27f9aac0-1c83-4d82-920c-dd028ed8853d)

I have opened an issue in `codesnake` (https://github.com/01mf02/codesnake/issues/2) to see if the creator wants the empty labels feature. Also, my work with codesnake is in the new `codesnake` branch of this repo.
